msprime/trees.py

```diff
@@ -289,17 +289,13 @@ def __init__(
         self._tree = tree
         self._x_coords = {}
         self._y_coords = {}
-        try:
-            self._branch_colours = {node: col for node, col in branch_colours.keys()}
-        except AttributeError:
-            self._branch_colours = {node: col for node, col in enumerate(branch_colours)}
-        
+        self._branch_colours = branch_colours or {}
```

Comment:
Following pep8, I think it's much clearer to explicitly compare with ``None`` when doing this sort of thing. Please change to
```python
self._branch_colours = {}
if branch_colours is not None:
    self._branch_colours = branch_colours
```
Ditto for node colours.